packages/otelbin/src/app/og/ParentsNode.tsx

```diff
@@ -29,61 +25,45 @@ const ParentsNode = ({ nodeData, nodes }: { nodeData: Node; nodes?: Node[] }) =>
 		?.filter((node) => node.type === "processorsNode")
 		.filter((processor) => processor.parentNode === nodeData.data.label);
 
-	const nodeHeight = 72;
-	const nodeTotalMargin = 40;
+	const parentNodeEdges = edges.filter((edge) => {
+		const sourceParent = edge.data.sourceParent;
+		const targetParent = edge.data.targetParent;
+		if (sourceParent === targetParent && sourceParent === nodeData.data.label) {
+			return edge;
+		}
+	});
 
-	function calcSVGHeight(nodes?: Node[]) {
-		const nodesCount = nodes?.length ?? 0;
-		return nodesCount * (nodeHeight + nodeTotalMargin) - nodeTotalMargin;
-	}
+	function drawSvgInsideParentNode(edges: Edge[], parentNode: Node) {
+		const padding = 10;
+		const nodeWidth = 120 + padding;
+		const halfNodeHeight = 80 / 2;
+		const edgesToDraw: IEdge[] = [];
+
+		if (!parentNode) {
+			return;
+		}
 
-	function calcSVGPath(side: string, nodes?: Node[]) {
-		const nodesCount = nodes?.length ?? 0;
-		const height = nodesCount * (nodeHeight + nodeTotalMargin) - 40;
+		edges.map((edge) => {
```

Comment:
**Nitpick:** Looks like we are ignoring the `map(…)` result. Use `forEach(…)` instead?

Actually, no, instead I'd suggest to return a value in the `map(…)` step and to remove `edgesToDraw`